test/autorest/mediatypesgroup/zz_generated_mediatypesclient.go

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+
+package mediatypesgroup
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+// MediaTypesClientOperations contains the methods for the MediaTypesClient group.
+type MediaTypesClientOperations interface {
+	// AnalyzeBody - Analyze body, that could be different media types.
+	AnalyzeBody(ctx context.Context, contentType ContentType, input azcore.ReadSeekCloser) (*StringResponse, error)
+	// AnalyzeBodyWithSourcePath - Analyze body, that could be different media types.
+	AnalyzeBodyWithSourcePath(ctx context.Context, mediaTypesClientAnalyzeBodyWithSourcePathOptions *MediaTypesClientAnalyzeBodyWithSourcePathOptions) (*StringResponse, error)
+	// ContentTypeWithEncoding - Pass in contentType 'text/plain; encoding=UTF-8' to pass test. Value for input does not matter
+	ContentTypeWithEncoding(ctx context.Context, input string) (*StringResponse, error)
+}
+
+// MediaTypesClient implements the MediaTypesClientOperations interface.
+// Don't use this type directly, use NewMediaTypesClient() instead.
+type MediaTypesClient struct {
+	*Client
+}
+
+// NewMediaTypesClient creates a new instance of MediaTypesClient with the specified values.
+func NewMediaTypesClient(c *Client) MediaTypesClientOperations {
+	return &MediaTypesClient{Client: c}
+}
+
+// Do invokes the Do() method on the pipeline associated with this client.
+func (client *MediaTypesClient) Do(req *azcore.Request) (*azcore.Response, error) {
+	return client.p.Do(req)
+}
+
+// AnalyzeBody - Analyze body, that could be different media types.
+func (client *MediaTypesClient) AnalyzeBody(ctx context.Context, contentType ContentType, input azcore.ReadSeekCloser) (*StringResponse, error) {
+	req, err := client.AnalyzeBodyCreateRequest(ctx, contentType, input)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	result, err := client.AnalyzeBodyHandleResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// AnalyzeBodyCreateRequest creates the AnalyzeBody request.
+func (client *MediaTypesClient) AnalyzeBodyCreateRequest(ctx context.Context, contentType ContentType, input azcore.ReadSeekCloser) (*azcore.Request, error) {
+	urlPath := "/mediatypes/analyze"
+	req, err := azcore.NewRequest(ctx, http.MethodPost, azcore.JoinPaths(client.u, urlPath))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", string(contentType))
+	req.Header.Set("Accept", "application/json")
```

Comment:
You might want to follow up with David about this to see if it's expected to add this header in the case where we are specifying a non-JSON payload.

Reply:
Will do, then add the update in another PR whenever he does get the fix in, if it will be fixed.

Reply:
Thinking more, this is likely an artifact of us making a copy of the original method.  If it doesn't hurt anything we can leave it as-is.